docs/site/todo-list-tutorial-sqldb.md

```diff
@@ -0,0 +1,153 @@
+steps in order to set up constraints in the database. We're using PostgreSQL for
+illustration, it would work for other relational databases in a similar way.
+
+### Create a new datasource
+
+We're going to reuse the DataSource `db` to connect to the PostgreSQL database
+instead of the in-memory database.
+
+First, remove the existing `db`, run:
+
+```sh
+rm src/datasources/db.datasource.*
+```
+
+Then, create the datasource with the same name with the `lb4 datasource` command
+and select `PostgreSQL` as the connector.
+
+```sh
+$ lb4 datasource
+? Datasource name: db
+? Select the connector for db:
+  Redis key-value connector (supported by StrongLoop)
+  MongoDB (supported by StrongLoop)
+  MySQL (supported by StrongLoop)
+❯ PostgreSQL (supported by StrongLoop)
+  Oracle (supported by StrongLoop)
+  Microsoft SQL (supported by StrongLoop)
+  REST services (supported by StrongLoop)
+  ...
+```
+
+### Specify the foreign key constraint in the models
+
+Based on the model relations we've specified in the 3 models, the expected
+database tables look like below:
+![database tables](todo-list-tutorial-dbtables.png)
+
+{% include note.html content="
+There is an ongoing work on supporting strong relation with referential integrity.  For details, please see [epic #2231](https://github.com/strongloop/loopback-next/issues/2331).
+" %}
+
+#### Specify the foreign key constraints in Todo model
+
+We will use the `foreignKeys` attribute to determine the constraints in the
+database table.
+
+In `src\models\todo.model.ts`, add the `settings` options in the `@model`
+decorator.
+
+```ts
+@model({
+  settings: {
+    foreignKeys: {
+      fk_todo_todoListId: {
+        name: 'fk_todo_todoListId',
+        entity: 'TodoList',
+        entityKey: 'id',
+        foreignKey: 'todolistid',
+      },
+    },
+  },
+})
+export class Todo extends Entity {
+```
+
+Check out the details for
+[auto-migrate and auto-update](https://loopback.io/doc/en/lb3/PostgreSQL-connector.html#auto-migrateauto-update-models-with-foreign-keys)
+for PostgreSQL connector. For other connectors, go to the corresponding
+connector from the
+[Connectors Reference](https://loopback.io/doc/en/lb3/Connectors-reference.html)
+section.
+
+#### Specify the foreign key constraints in TodoListImage model
+
+Similar to the `Todo` model, we will specify the constraints in the `settings`.
+
+```ts
+@model({
+  settings: {
+    foreignKeys: {
+      fk_todo_todoListId: {
+        name: 'fk_todo_todoListId',
+        entity: 'TodoList',
+        entityKey: 'id',
+        foreignKey: 'todolistid',
+      },
+    },
+  },
+})
+export class TodoListImage extends Entity {
+```
+
+### Database migration using `npm run migrate` command
+
+The order of table creation is important. We are going to migrate `TodoList`
+model first and then `Todo` and `TodoListImage` models.
+
+1. We'll run the database migration on the `TodoList` table. In
```

Comment:
I am wondering...why we don't run the migrate command only once with config `models: ['TodoList', 'Todo', 'TodoListImage']`?

Reply:
Would the migration happen in this order? If yes, that's even better. let me give it a try. thanks.

Reply:
it is working!  I've fixed my docs accordingly.